frontend/app/components/sections/Landingpage.jsx

```diff
@@ -24,46 +27,70 @@ export default function Landingpage() {
       return;
     }
 
-    toast.promise(sendFileToBackend(file, outputFormat), {
-      pending: {
-        render: data.landing.uploading,
-        position: "bottom-right",
-        autoClose: false,
-        hideProgressBar: false,
-      },
-      success: {
-        render: data.landing.uploaded,
-        position: "bottom-right",
-        autoClose: 2000,
-        hideProgressBar: true,
-      },
-      error: {
-        render: data.landing.uploadError,
-        position: "bottom-right",
-        autoClose: 2000,
-        hideProgressBar: true,
-      },
-    });
+    try {
+      const { details } = await toast.promise(
+        sendFileToBackend(file, outputFormat),
+        {
+          pending: {
+            render: data.landing.uploading,
+            position: "bottom-right",
+            autoClose: false,
+            hideProgressBar: false,
+          },
+          success: {
+            render: data.landing.uploaded,
+            position: "bottom-right",
+            autoClose: 2000,
+            hideProgressBar: true,
+          },
+          error: {
+            render: data.landing.uploadError,
+            position: "bottom-right",
+            autoClose: 2000,
+            hideProgressBar: true,
+          },
+        },
+      );
+
+      setDetails(details);
+      setShowModal(true);
+    } catch (error) {
+      throw error;
+    }
+  };
+
+  const closeModal = () => {
+    setShowModal(false);
+  };
+
+  const openModal = () => {
+    setShowModal(true);
   };
 
   return (
     <main>
-      <div className="flex flex-col gap-5 justify-center items-center pt-10">
+      <div className="flex flex-col gap-3 justify-center items-center pt-10">
         <FileUpload onFileUpload={handleFileUpload} />
-        <div className="text-3xl text-center font-bold text-blue pt-10">
-          <p>{data.landing.download}</p>
+        <Button onClick={openModal} variant="primary">
+          {data.landing.details}
```

Comment:
It is a little bit confusing that when you "upload" the file and don't click on "Moodle" or "Coursera" buttons. The "Details" button can be clicked and shows everything as null. It looks like the "uploaded" file is empty.
![image](https://github.com/AnneliKlamas/test-converter/assets/44496463/9b4cd018-6643-4c2e-8471-007edfad7f4f)
![image](https://github.com/AnneliKlamas/test-converter/assets/44496463/18cd43b1-256a-4ab9-b0ad-a2a4c4a44fee)